### ISSUE TITLE
Unused parameter in init_scheduler() of scheduler config classes

### ISSUE THREAD

**Issue by hieuchi911** [fostiropoulos/ablator#16]
1. The method `init_scheduler()` in all scheduler config classes are declaring `model: nn.Module` as a parameter, but this is not used when initializing the schedulers. Changing it to **kwargs or *args might be a reasonable solution
2. `optimizer` parameter should be of type `Optimizer`

**Comment by hieuchi911** [comment]
In `examples`, 2 is addressed but 1 is not addressed yet

**Comment by fostiropoulos** [comment]
As discussed some schedulers might require a model for an internal operation. This issue is not possible to address as it requires specifically passing a model to the scheduler from the wrapper.
